### ISSUE TITLE
Link resumption does not carry unsettled deliveries (cross-session drops them, same-session resume fails)

### ISSUE THREAD

**Issue by minghuaw** [minghuaw/fe2o3-amqp#396]
## Summary

Unsettled deliveries do not survive link resumption:

- **Cross-session/connection resume** (`detach_then_resume_on_session` / `DetachedSender::resume_on_session`): the resume succeeds, but the pending delivery is neither advertised nor re-sent; the `DeliveryFut` resolves with `SendError::LinkStateError(SessionStopped(..))`.
- **Same-session resume** (`detach()` + `DetachedSender::resume()`): the delivery *is* re-sent (the peer receives it), but `resume()` returns `SenderResumeError { kind: AttachError(IllegalState) }` and the delivery future never resolves.

The intended behavior (API says "resume the link"; `send_batchable` returns a future that does not borrow the sender) is that unsettled deliveries are carried over and their futures resolve with the peer's disposition after resumption.

### Cross-session

`Link::send_attach_inner` clears the local unsettled map when `is_reattaching` (link moved to a different session/connection):

```rust
// fe2o3-amqp/src/link/mod.rs
let unsettled_map_len = if is_reattaching {
    let mut guard = self.unsettled.write();
    *guard = None;   // drops the UnsettledMessage and its settlement oneshot
    None
} else { ... };
```

`Link::get_unsettled_map` also returns `None` when `is_reattaching`, so the attach advertises no unsettled map. Dropping the map drops each delivery's settlement sender, so `DeliveryFut::poll` takes the `Err(recv)` path and reports `from_session_stop_reason` (`SessionStopped`). No `resend` happens.

### Same-session

`SenderInner::resume_incoming_attach` handles `SenderAttachExchange::IncompleteUnsettled` by processing the deliveries but not completing the exchange:

```rust
SenderAttachExchange::IncompleteUnsettled(resuming_deliveries) => {
    for (delivery_tag, resuming) in resuming_deliveries {
        self.handle_resuming_delivery(delivery_tag, resuming, &mut resend_buf).await?;
    }
    // no detach/break
}
SenderAttachExchange::Resume(resuming_deliveries) => {
    ...
    self.detach_with_error(None).await?;   // reduces state, then the loop reattaches
}
```

The loop then calls `exchange_attach` again with the link in `IncompleteAttachExchanged`, which `send_attach_inner` rejects (`_ => IllegalState`), so `resume()` fails. (The peer did receive the re-sent delivery before the failure.)

## Reproduction

Probe over `tokio::io::duplex` (no broker):

1. Establish two connection/session pairs A and B.
2. Attach a sender on A (acceptor receiver on the listener side).
3. `let delivery = sender.send_batchable("hello").await?;` (unsettled transfer on the wire).
4. Cross-session: drop session A/connection A, accept the reattach on B, `sender.detach_then_resume_on_session(&session_b)`.
   - observed: resume `Ok(())`, peer `recv` times out, `delivery` -> `Err(LinkStateError(SessionStopped(Ended)))`.
5. Same-session: drive the acceptor receiver so it answers a non-closing detach, `sender.detach()`, accept the reattach on the same session, `detached.resume()`.
   - observed: peer receives `"hello"`, resume -> `Err(SenderResumeError { kind: AttachError(IllegalState) })`, `delivery` never resolves.

## Expected

- The pending delivery survives the session/connection stop and is re-sent after resumption; `send_batchable`'s future resolves with the peer's disposition.
- `DetachedSender::resume()` / `resume_on_session()` complete successfully when there are unsettled deliveries.

## Fix sketch

- Cross-session: do not clear the unsettled map in `send_attach_inner`; advertise it on reattach (`get_unsettled_map` should not return `None` just because `is_reattaching`) so the peer can reconcile, and re-send via the existing `ResumingDelivery` path.
- Same-session: make the `IncompleteUnsettled` arm complete the exchange like `Resume` (settle the reduced state, detach, reattach) instead of looping into `exchange_attach` from `IncompleteAttachExchanged`.

Found while adding resumption docs/tests for the error-type refactor (branch `refactor/error-types`).
